app/lib/features/pins/pages/pin_details_page.dart

```diff
@@ -0,0 +1,286 @@
+        actions.add(
+          PopupMenuItem<String>(
+            onTap: () => showReportDialog(context, pin),
+            child: Row(
+              children: <Widget>[
+                Icon(
+                  Atlas.warning_thin,
+                  color: Theme.of(context).colorScheme.error,
+                ),
+                const SizedBox(width: 10),
+                Text(L10n.of(context).reportPin),
+              ],
+            ),
+          ),
+        );
+      }
+    }
+
+    return actions;
+  }
+
+  Widget _buildPinHeaderUI(ActerPin pin) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 16),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Row(
+            children: [
+              pinIconUI(),
+              const SizedBox(width: 12),
+              Expanded(
+                child: Column(
+                  mainAxisSize: MainAxisSize.min,
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    pinTitleUI(pin),
+                    pinSpaceNameUI(pin),
+                  ],
+                ),
+              ),
+            ],
+          ),
+          pinDescriptionUI(pin),
+        ],
+      ),
+    );
+  }
+
+  Widget pinIconUI() {
+    return Container(
+      height: 50,
+      width: 50,
+      decoration: BoxDecoration(
+        color: Theme.of(context).colorScheme.secondary,
+        borderRadius: const BorderRadius.all(Radius.circular(100)),
+      ),
+      child: const Icon(Atlas.pin),
+    );
+  }
+
+  Widget pinTitleUI(ActerPin pin) {
+    return SelectionArea(
+      child: GestureDetector(
+        onTap: () {
+          final membership =
+              ref.watch(roomMembershipProvider(pin.roomIdStr())).valueOrNull;
+          if (membership != null && membership.canString('CanPostPin')) {
+            showEditTitleBottomSheet(
+              context: context,
+              bottomSheetTitle: L10n.of(context).editName,
+              titleValue: pin.title(),
+              onSave: (newTitle) async {
+                final pinEditNotifier =
+                    ref.watch(pinEditProvider(pin).notifier);
+                pinEditNotifier.setTitle(newTitle);
+                savePinTitle(context, pin, newTitle);
+              },
+            );
+          }
+        },
+        child: Text(
+          pin.title(),
+          style: Theme.of(context).textTheme.titleSmall,
+        ),
+      ),
+    );
+  }
+
+  Widget pinSpaceNameUI(ActerPin pin) {
+    return SpaceChip(
+      spaceId: pin.roomIdStr(),
+      useCompatView: true,
+    );
+  }
+
+  Widget pinDescriptionUI(ActerPin pin) {
+    final description = pin.content();
+    if (description == null) return const SizedBox.shrink();
+    final formattedBody = description.formattedBody();
+
+    if (formattedBody == null && description.body().trim().isEmpty) {
```

Comment:
avoid multiple reads of the same value from the api... it is slow and inefficient.
```suggestion
    final body = description.body();
    if (formattedBody == null && body.trim().isEmpty) {
```